tools/np-build-driver-stride.lua

```diff
@@ -0,0 +1,113 @@
+ffi = require 'ffi'
+
+local SLASH = "/"
+if ffi.os == "Windows" then SLASH = "\\" end
+
+-- Code by David Kastrup
+require "lfs"
+
+function dirtree(dir)
+  assert(dir and dir ~= "", "directory parameter is missing or empty")
+  if string.sub(dir, -1) == SLASH then
+    dir=string.sub(dir, 1, -2)
+  end
+
+  local function yieldtree(dir)
+    for entry in lfs.dir(dir) do
+      if entry ~= "." and entry ~= ".." then
+        entry=dir..SLASH..entry
+        local attr=lfs.attributes(entry)
+        coroutine.yield(entry,attr)
+        if attr.mode == "directory" then
+          yieldtree(entry)
+        end
+      end
+    end
+  end
+
+  return coroutine.wrap(function() yieldtree(dir) end)
+end
+
+function string.ends(String,End)
+   return End=='' or string.sub(String,-string.len(End))==End
+end
+
+function string.starts(String,Start)
+   return string.sub(String,1,string.len(Start))==Start
+end
+
+local exclude_dirs = {}
+local exclude_files = {}
+local directory = ""
+
+-- We may need to support both build approaches for now (iOS / android)
+common_flags = "-ffast-math -fno-threadsafe-statics -nostdlibinc -nobuiltininc -nostdinc++ -Wno-macro-redefined -DFLOAT_APPROX -I.."..SLASH.."NativePath -I.."..SLASH.."NativePath"..SLASH.."standard"
+zig_common_flags = "-Wno-ignored-attributes -Wno-delete-non-virtual-dtor -Wno-macro-redefined -fno-rtti -fno-exceptions"
+
+objs = {}
+cfiles = {}
+cppfiles = {}
+hfiles = {}
+debug = false
+is_verbose = false
+is_shared = false
+platform = "windows"
+outputName = "lib"
+
+function table.contains(table, element)
```

Comment:
Can you move generic helpers like this, the string ones above, dirtree and Slash choosing to another file and only keep custom build related logic here?

Reply:
I could yes, but I really just wanted to leave the original impl alone, and so I created these complete duplicates so I could apply the cleanup to just what Stride actually needs. Keeping the old intact is useful if / when we need to review it for other runtime support. And I think mobile will still need to use some of that old approach. But not sure yet.

I've already spent way too much time on this stuff, and don't want to risk breaking anything more, so if you guys are ok with it I'd prefer not to refactor this too much, for minimal value.

Reply:
Ok I've just eliminated the stride specific driver file, and moved the few small changes into the common shared one now.